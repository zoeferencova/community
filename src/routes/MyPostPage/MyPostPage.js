import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import moment from "moment";
import tz from "moment-timezone";
import CommUnityContext from "../../contexts/context";
import UserDataService from "../../services/user-data-service";
import Task from "../../components/Task/Task"
import GoogleMap from "../../components/GoogleMap/GoogleMap";
import { ButtonLight, ButtonDark } from "../../components/Utils/Utils";
import styles from "./MyPostPage.module.css";

class MyPostPage extends Component {
    static contextType = CommUnityContext;

    state = {
        loading: false,
    }
    
    findPost() {
        const postId = parseFloat(this.props.match.params.id);
        const post = this.context.user_posts.find(post => post.id === postId)
        return post;
    }

    handleDelete(postId) {
        if (window.confirm("Are you sure you want to delete this post?")) {
            this.setState({ loading: true })
            UserDataService.deletePost(postId)
                .then(res => {
                    this.setState({ loading: false })
                    this.props.history.push("/home")
                    this.context.removePost(postId)
                })
        }
    }
    
    render() {
        const post = this.findPost()
        return (   
            <main className={styles.main}>
                {post && <>
                <div className={styles.map}>
                    <GoogleMap location={post.location} radius={parseFloat(post.radius)} displayMarker={true} />                
                </div>
                <div className={styles.postHeader}>
                    <div>
                        <h3>Your {post.post_type}</h3>
                        <p className={styles.date}>{moment(post.date_created).tz(this.context.timeZone).format('LLLL z')}</p>
                    </div>
                    {post.post_type === "request" && <span className={`${styles[post.urgency]} ${styles.urgency}`}><i className="fas fa-circle"></i> {post.urgency} urgency</span>}
                </div>
                <ul className={styles.tasks}>
                    {post.categories.map(task => <Task key={task} task={task} />)}
                </ul>
                {post.description && <p className={styles.description}>{post.description}</p>}
                <div className={styles.buttonSection}>
                    <ButtonLight type="button" onClick={() => this.props.history.push("/home")}>Back</ButtonLight>
                    <div className={styles.rightButtons}>
                        <ButtonLight className={styles.delete} onClick={() => this.handleDelete(post.id)} loading={this.state.loading.toString()}>Delete</ButtonLight>
                        <ButtonDark type="button" onClick={() => this.props.history.push(`/edit-post/${post.id}`)}>Edit</ButtonDark>
                    </div>
                </div>
                </>}
            </main>
        )
    }
}

export default withRouter(MyPostPage)