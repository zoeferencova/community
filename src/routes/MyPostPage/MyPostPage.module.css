.main {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0 auto;
    position: absolute;
    top: 65px;
    padding: 30px;
}

.tasks {
    display: flex;
    list-style: none;
    margin: 0px;
    padding-left: 0px;
}

.map {
    height: 250px;
    width: 100%;
    background-color: #ddd;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 12px;
    margin-bottom: 15px;
}

@media only screen and (min-width: 600px) {
    .map {
        width: 400px;
    }
}