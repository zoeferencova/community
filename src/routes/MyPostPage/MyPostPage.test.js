import React from "react";
import { render } from "enzyme";
import MyPostPage from "./MyPostPage";
import { MemoryRouter, Route } from "react-router-dom";
import CommUnityContext from "../../contexts/context";

import mockContext from "../../contexts/mock-context";

describe("MyPostPage", () => {
    it("renders MyPostPage component given mock context data", () => { 
        const contextCopy = {...mockContext}
        contextCopy.user_posts.forEach(post => post.date_created = "2020-10-08T16:36:52-04:00")
        const wrapper = render(
            <CommUnityContext.Provider value={contextCopy}>
                <MemoryRouter initialEntries={[`/my-post/1`]}>
                    <Route path="/my-post/:id">
                        <MyPostPage />
                    </Route>
                </MemoryRouter>
            </CommUnityContext.Provider>,
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});