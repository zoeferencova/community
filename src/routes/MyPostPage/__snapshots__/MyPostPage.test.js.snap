// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPostPage renders MyPostPage component given mock context data 1`] = `
<main
  class="main"
>
  <div
    class="map"
  >
    <div
      id="google-map"
      style="height:100%;width:100%"
    />
  </div>
  <div
    class="postHeader"
  >
    <div>
      <h3>
        Your request
      </h3>
      <p
        class="date"
      >
        Thursday, October 8, 2020 4:36 PM EDT
      </p>
    </div>
    <span
      class="low urgency"
    >
      <i
        class="fas fa-circle"
      />
       low urgency
    </span>
  </div>
  <ul
    class="tasks"
  >
    <li
      class="task call"
    >
      Phone call
    </li>
  </ul>
  <div
    class="buttonSection"
  >
    <button
      class="button "
      type="button"
    >
      Back
    </button>
    <div
      class="rightButtons"
    >
      <button
        class="button delete"
        loading="false"
      >
        Delete
      </button>
      <button
        class="buttonDark "
        type="button"
      >
        Edit
      </button>
    </div>
  </div>
</main>
`;
